src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -122,6 +122,28 @@ class PrisonerIepLevelReviewService(
     }
   }
 
+  private suspend fun getIepLevelForReviewType(prisonerInfo: PrisonerAtLocation, reviewType: ReviewType): String {
+    val iepLevelsForPrison = iepLevelService.getIepLevelsForPrison(prisonerInfo.agencyId)
+    when (reviewType) {
+      ReviewType.INITIAL -> {
+        return iepLevelsForPrison.first(IepLevel::default).iepLevel
+      }
+
+      ReviewType.TRANSFER -> {
+        val currentIepLevelForPrisoner = getCurrentIEPLevelForPrisoners(bookingIds = listOf(prisonerInfo.bookingId), useNomisData = true, useClientCredentials = true)
+          .firstOrNull() ?: throw NoDataFoundException(prisonerInfo.bookingId)
+        return iepLevelsForPrison.find { it.iepDescription == currentIepLevelForPrisoner.iepLevel }
+          ?.iepLevel
+          ?: run {
```

Comment:
is this `?: run` some kind of ternary operator (if/else)?

I'm guessing this means:

> if we can find the current prisoner level in the prison levels
>    => return this level
> else
>    => return the highest level for the prison

Is that correct? Sorry this is a bit terse, I should have a look at the Kotlin syntax.

Reply:
Exactly. It's the [Elvis operator](https://kotlinlang.org/docs/null-safety.html#elvis-operator). `find` could return null and so we want to handle that scenario with the logic you described in the quote block

I did think about extracting `iepLevelsForPrison.find { it.iepDescription == currentIepLevelForPrisoner.iepLevel }` into a variable - something like `prisonSupportsPrisonerIepLevel` - but wasn't sure if it would help with readability